modules/jpa/src/main/java/io/oasp/module/jpa/common/base/AbstractGenericDao.java

```diff
@@ -0,0 +1,401 @@
+package io.oasp.module.jpa.common.base;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityNotFoundException;
+import javax.persistence.LockModeType;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+import net.sf.mmm.util.entity.api.PersistenceEntity;
+import net.sf.mmm.util.exception.api.ObjectNotFoundUserException;
+import net.sf.mmm.util.search.base.AbstractSearchCriteria;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.querydsl.core.types.Expression;
+import com.querydsl.jpa.impl.JPAQuery;
+
+import io.oasp.module.basic.common.api.reference.Ref;
+import io.oasp.module.jpa.common.api.to.PaginatedListTo;
+import io.oasp.module.jpa.common.api.to.PaginationResultTo;
+import io.oasp.module.jpa.common.api.to.PaginationTo;
+import io.oasp.module.jpa.common.api.to.SearchCriteriaTo;
+import io.oasp.module.jpa.dataaccess.api.GenericDao;
+
+/**
+ * This is the abstract base-implementation of the {@link GenericDao} interface.
+ *
+ * @param <ID> is the generic type if the {@link PersistenceEntity#getId() primary key}.
+ * @param <E> is the generic type of the managed {@link PersistenceEntity}.
+ *
+ * @deprecated It is only provided to support old functionality of DAO, instead use
+ *             io.oasp.module.jpa.dataaccess.base.AbstractGenericDao instead
+ */
+// @Repository
+@Deprecated
+public abstract class AbstractGenericDao<ID, E extends PersistenceEntity<ID>> implements GenericDao<ID, E> {
```

Comment:
This is also fine in general but I would expect this to extend another abstract class from `oasp4j-jpa-dao` containing the general DAO implementation stuff and only having the legacy support methods (paging support, etc.) here.

Reply:
This can be done all Paging support method put in separate class